portal/app/Http/Controllers/PortalController.php

```diff
@@ -47,11 +47,22 @@ public function authenticate(Request $request, \UniFi_API\Client $unifi)
             return $this->status();
         }
 
-        $this->unifi->authorize_guest($client->mac, 0);
+        // TODO: configuration of duration
+        // we assume we are running a weekend long, 3 days
+        $minutes = 60*24*3;
+        $this->unifi->authorize_guest($client->mac, $minutes);
 
         $voucher->used_at = now();
+        $voucher->mac = $client->mac;
         $voucher->save();
 
+        // also attempt to update the name in the controller
+        $participant = $voucher->participant()->get();
+        if (count($participant) > 0) {
+            $participant = $participant[0];
+            $this->unifi->set_sta_name($client->user_id, $participant->name);
+        }
```

Comment:
Since participant is defined as a BelongsTo relation, the following would theoretically be sufficient:

```
if ($participant = $voucher->participant) {
    $this->unifi->set_sta_name($client->user_id, $participant->name);
}
```